### ISSUE TITLE
[Parser] Hyphen in participant name bring TypeError

### ISSUE THREAD

**Issue by sudodoki** [knsv/mermaid#74]
If you have participant, whose name contain `-`, in sequence diagram, mermaid will fail with following message
```
TypeError: 'undefined' is not an object (evaluating 'action[0]')

  ../dist/mermaid.full.js:15352 in parse
  ../dist/mermaid.full.js:16236
  ../dist/mermaid.full.js:16466
  ../dist/mermaid.full.js:16517
  phantomjs://webpage.evaluate():23 in executeInPage
  phantomjs://webpage.evaluate():29
  phantomjs://webpage.evaluate():29
PHANTOM ERROR: TypeError: 'null' is not an object (evaluating 'element.setAttribute')
TRACE:
 -> /usr/local/lib/node_modules/mermaid/lib/phantomscript.js: 149 (in function resolveSVGElement)
 -> /usr/local/lib/node_modules/mermaid/lib/phantomscript.js: 69%
```
Mermaid version is 0.3.0
Phantomjs version is 1.9.8

**Comment by knsv** [comment]
Been looking at this and it is a bit tricky to fix as the Hyphen is also involved in the signal type. It is possible to fix but would make the grammar much more complex. Perhaps we could encode it somehow?

**Comment by knsv** [comment]
Fix for this issue was released in mermaid 0.5.2. Will close this now, let me know if you issues still remain.

**Comment by sudodoki** [comment]
Thanks!
